src/validator.ts

```diff
@@ -680,6 +682,51 @@ function doIntrinsicGetAZs(ref: any, key: string){
 
 }
 
+function doIntrinsicSelect(ref: any, key: string){
+  let toGet = ref[key];
+  if(!Array.isArray(toGet) || toGet.length < 2) {
+    addError('crit', "Fn::Select only supports an array of two elements", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[0] === undefined || toGet[0] === null) {
+     addError('crit', "Fn::Select first element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let index = parseInt(toGet[0]);
+  if (typeof index !== 'number' || isNaN(index)) {
+    addError('crit', "First element of Fn::Select must be a number", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[1] === undefined || toGet[1] === null) {
+     addError('crit', "Fn::Select Second element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let list = toGet[1]
+  if (!Array.isArray(list)) {
+    //we may need to resolve it
+    if (typeof list !== 'string') {
+      list = resolveIntrinsicFunction(list, Object.keys(list)[0]);
+    } else {
+      addError('crit', "Fn::Select requires the second element to resolve to a list, it appears to be a string", placeInTemplate, "Fn::Select");
+      return 'INVALID_SELECT';
+    }
+    if (!Array.isArray(list)) {
+      addError('crit', "Fn::Select requires the second element to be a list, function call did not resolve to a list", placeInTemplate, "Fn::Select");
+      return 'INVALID_SELECT';
+    }
+  }
+  if (index >= 0 && index < list.length) {
+    return list[index];
+  } else {
+    addError('warn', "First element of Fn::Select exceeds the length of the list, if list is a function, make sure it returns a longer list", placeInTemplate, "Fn::Select");
```

Comment:
This should probably result in a error instead as it will fail the deployment.
> Fn::Select does not check for null values or if the index is out of bounds of the array. Both conditions will result in a stack error, so you should be certain that the index you choose is valid, and that the list contains non-null values.

Reply:
I was unsure on that one. Since it was possible that a mocked intrinsic could have supplied a shorter list than would be was expected. Fn::Select I think is recommended for choosing AZs, but it looked like the mocked getAZ was only returning three entries. (I'm not sure about that, I only had a minute to glance over it. - and I'm definitely not an expert that front :) )

Reply:
Yeah, I was wondering about that too. It would involve lots of effort to improve the getAZ mock (i.e. integration with the AWS SDK), as you can't guarantee that all account have the same number of AZs in each region. 

> Therefore, your account might have a different number of available Availability Zones in a region than another account.

We could, however, have a flag to define the amount of AZs you wish to validate against, as we can always expect this to start from _a_, AFAIK.

If you can think of any other edge cases where making this an "error" would cause annoyance then we might need to think of another solution.